src/js/jsx/sections/layers/LayersPanel.jsx

```diff
@@ -467,7 +467,7 @@ define(function (require, exports, module) {
 
                 var dragSource = collection.pluck(this.state.dragTargets, "id");
 
-                flux.actions.layers.reorder(doc, dragSource, dropIndex)
+                return flux.actions.layers.reorder(doc, dragSource, dropIndex)
```

Comment:
OK, I understand a little better. I agree that the `finally` handler makes sense here, but I still think the `then` handler should be part of the action.